### PR TITLE
Improve documentation for PointerSignalResolver class

### PR DESCRIPTION
**Pull request by tvolkert**
Make it more discoverable in the docs, and go into a little more depth on why the class is needed and how to use it.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by tvolkert** [comment]
Test exemption: documentation-only change.

File: packages/flutter/lib/src/gestures/events.dart [hunk 1]
```diff
@@ -1847,6 +1847,9 @@ class _TransformedPointerUpEvent extends _TransformedPointerEvent with _CopyPoin
 ///
 ///  * [Listener.onPointerSignal], which allows callers to be notified of these
 ///    events in a widget tree.
+///  * [PointerSignalResolver], which ensures that these events will only be
```
Comment (Hixie) [inline packages/flutter/lib/src/gestures/events.dart:1850]: it doesn't ensure that, right? It only provides an opt-in mechanism for cooperating agents to disambiguate an event's target.

File: packages/flutter/lib/src/gestures/events.dart [hunk 2]
```diff
@@ -1916,6 +1919,9 @@ mixin _CopyPointerScrollEvent on PointerEvent {
...
+///    dispatched to one registered handler when multiple objects are
+///    interested in the event.
```
Comment (Hixie) [inline packages/flutter/lib/src/gestures/events.dart:1924]: ditto

File: packages/flutter/lib/src/gestures/pointer_signal_resolver.dart [hunk 3]
```diff
@@ -15,22 +15,38 @@ bool _isSameEvent(PointerSignalEvent event1, PointerSignalEvent event2) {
   return (event1.original ?? event1) == (event2.original ?? event2);
 }
 
-/// An resolver for pointer signal events.
+/// An object that mediates disputes over which listener should handle pointer
```
Comment (Hixie) [inline packages/flutter/lib/src/gestures/pointer_signal_resolver.dart:18]: "An object that" is redundant

**Review by Hixie — APPROVED** [review]
LGTM.

Bonus points for an interactive code sample.

**Pull request merged**